### PR TITLE
fix: added loading state for chart widget

### LINKED ISSUE #10183: [Bug]: Chart widget, Not in loading state when the query is still loading data
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The chart widget is not in the loading state when the data from the query is still loading, The UX for the end user is confusing as then dont know if there is an error in showing data, specially in case of large data set.

Issue was originally reported on [discord](https://discord.com/channels/725602949748752515/760761686549463060/928231994406625330)

### Steps To Reproduce

1. Add a chart widget
2. Use a query to load data, transform it to what the widget expects
3. you will see that the loading state does not come up
    1.  I have even tried setting the loading property in the widget with the query.isloading property. Still doesnt work

[sample app](https://release.app.appsmith.com/applications/60bf75b7ce05af69b4848b14/pages/60bf75b7ce05af69b4848b16/edit)

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by keyurparalkar**
## Description

**Summary of changes**

- Added class `bp3-skeleton` when the `isLoading` prop is true.

**Motivation**

- A loading state did not existed when `Animate Loading` was set to true. This caused the widget to crash whenever the widget was attached with the query data. 

Fixes #10183

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

This fix is tested manually. Please find below the steps to test the fix:

- Add a chart widget on to the canvas.
- Attach a query to the chart's `Series Data` attribute for example, `{{Query1.data.map( (obj) =>{ return  {'x': obj.gender, 'y': obj.id } })}}`
- Set the property `Animate Loading` to true.
- Refresh the page.
- Observe that you will see a loader.

Please find below video for reference:

[LOOM DEMO](https://www.loom.com/share/b9bf02ee25b34e47ba1460a8decc8462)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/fix-chart_widget_no_loading_state 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | app/client/src/widgets/ChartWidget/component/index.tsx | 49.41 **(0)** | 22.68 **(0.57)** | 50 **(0)** | 47.47 **(0)**</details>

### CODE REVIEW

**Comment by keyurparalkar** [comment]
Documentation update: https://github.com/appsmithorg/appsmith-docs/pull/321

**Comment by somangshu** [comment]
> Documentation update: [appsmithorg/appsmith-docs#321](https://github.com/appsmithorg/appsmith-docs/pull/321)

cc @dilippitchika @Pranay105

**Comment by somangshu** [comment]
@keyurparalkar can we cover this with test?

**Review by somangshu — APPROVED** [review]
Change LGTM

**Comment by somangshu** [comment]
Tested the DP, and it looks good to me.

The only thing to note is that the query isloading is only true on page load, while rerunning the query the property remains false, is that expected? @hetunandu @riodeuno 
Screenshot 2022-01-12 at 12 22 21 PM

File: app/client/yarn.lock [hunk 1]
```diff
@@ -6248,6 +6248,11 @@ cypress-real-events@^1.5.1:
   version "1.5.1"
   resolved "https://registry.yarnpkg.com/cypress-real-events/-/cypress-real-events-1.5.1.tgz#5eeb86d2a7aad9aa6d5271e288a23e46373915cd"
 
+cypress-wait-until@^1.7.2:
```
Comment (somangshu) [inline app/client/yarn.lock:6251]: what is this for, can we get this confirmed from @yatinappsmith, i.e. if this is an update in any sense

**Review by somangshu — APPROVED** [review]
Change LGTM

File: app/client/yarn.lock (hunk 1 above)
Comment (keyurparalkar) [reply app/client/yarn.lock:6251]: @somangshu I confirmed this with @Aishwarya-U-R. This is an expected change

**Pull request merged**